lib/Sema/CSSimplify.cpp

```diff
@@ -2869,47 +2869,55 @@ getArgumentLabels(ConstraintSystem &cs, ConstraintLocatorBuilder locator) {
 /// particularly fast in the face of deep class hierarchies or lots of protocol
 /// conformances, but this is fine because it doesn't get invoked in the normal
 /// name lookup path (only when lookup is about to fail).
-static bool hasDynamicMemberLookupAttribute(Type ty) {
-  // If this is a protocol composition, check to see if any of the protocols
-  // have the attribute on them.
-  if (auto protocolComp = ty->getAs<ProtocolCompositionType>()) {
-    for (auto p : protocolComp->getMembers())
-      if (hasDynamicMemberLookupAttribute(p))
-        return true;
-    return false;
-  }
+static bool hasDynamicMemberLookupAttribute(Type ty,
+                       llvm::DenseMap<Type, bool> &IsDynamicMemberLookupCache) {
```

Comment:
I suggest making the input a `CanType`, and I guess (annoyingly) we shouldn't cache anything is `ty->hasTypeVariable()`. Sorry for not making this clearer earlier.